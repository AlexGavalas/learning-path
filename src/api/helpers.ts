import axios from 'axios';

export const fetchFileFromStorage = async (
    filename: string,
): Promise<string> => {
    const fileServerUrl = String(process.env.PUBLIC_FILE_SERVER_URL);

    const response = await axios.get(`${fileServerUrl}/${filename}`, {
        responseType: 'text',
    });

    return response.data as string;
};
