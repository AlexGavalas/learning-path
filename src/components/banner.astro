<p class="text-center">
    Hey there. ðŸ‘‹ This is a place where I keep all the interesting stuff I come
    across. Feel free to explore. It is a WIP, so you can expect things to
    change.
</p>
