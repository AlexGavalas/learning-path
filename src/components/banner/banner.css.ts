import { style } from '@vanilla-extract/css';

export const banner = style({
    textAlign: 'center',
});
