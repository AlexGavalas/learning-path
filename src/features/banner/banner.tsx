export const Banner = () => (
    <p className="p-2 text-center">
        Hey there. ðŸ‘‹ This is a place where I keep all the interesting stuff I
        come across. Feel free to explore. It is a WIP, so you can expect things
        to change.
    </p>
);
