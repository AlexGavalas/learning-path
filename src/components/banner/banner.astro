---
import * as cn from './banner.css';
---

<p class={cn.banner}>
    <slot />
</p>
