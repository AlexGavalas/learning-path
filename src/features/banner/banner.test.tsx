import { axe } from 'jest-axe';

import { renderWithUser } from '~test/helpers';
import { Banner } from './banner';

describe('<Banner />', () => {
    it('renders', () => {
        const { container } = renderWithUser(<Banner />);

        expect(container).toMatchSnapshot();
    });

    it('is accessible', async () => {
        const { container } = renderWithUser(<Banner />);

        const a11yResults = await axe(container);

        expect(a11yResults).toHaveNoViolations();
    });
});
