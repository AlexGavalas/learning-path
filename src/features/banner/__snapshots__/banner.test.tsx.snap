// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> renders 1`] = `
<div>
  <p
    class="m-0 p-2 text-center"
  >
    Hey there. ðŸ‘‹ This is a place where I keep all the interesting stuff I come across. Feel free to explore. It is a WIP, so you can expect things to change.
  </p>
</div>
`;
